### PR TITLE
fix:failed to enable two batch overlap when use pp

### PR DESCRIPTION
**Pull request by Joey-gvwal**
## Motivation

When using the two-batch overlap (TBO) feature, EP (Expert Parallelism) + DP (Data Parallelism) is typically employed. However, some scenarios require PP (Pipeline Parallelism) due to hardware constraints. Because PP parallelism is used, pp miss layers are generated. When entering the TBO logic, it fails with an error when trying to read layer-related attributes:

```
sglang/python/sglang/srt/two_batch_overlap.py", line 803, in model_forward_maybe_tbo
    layer_input_scatter_mode = layers[0].layer_scatter_modes.layer_input_mode
  File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1931, in __getattr__
    raise AttributeError(
```

This PR provides a fix by adding layer filtering before entering TBO logic, filtering out pp miss layers.

## Modifications

In the model_forward_maybe_tbo function, check whether the layer is a pp-miss layer; if so, filter it out and return the real layer for forward execution.

```
def filter_real_layers(layers: nn.ModuleList) -> List[nn.Module]:
        return [l for l in layers if type(l).__name__ != "PPMissingLayer"]
real_layers = filter_real_layers(layers)
```

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).

### CODE REVIEW

**Comment by Joey-gvwal** [comment]
@yizhang2077 @zhyncs @fzyzcjy cloud you please take a look?

**Review by fzyzcjy — APPROVED** [review]
LGTM, could you please fix lint

**Comment by Joey-gvwal** [comment]
@fzyzcjy  I have fixed the lint errors, but I don’t have permission to add the run-ci label. Could you please add it for me?
